builtin/credential/approle/path_role.go

```diff
@@ -521,8 +521,26 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 	return logical.ListResponse(listItems), nil
 }
 
-// setRoleEntry grabs a write lock and stores the options on an role into the storage.
-// Also creates a reverse index from the role's RoleID to the role itself.
+// validRoleConstraints tells if the role has at least one constraint enabled
+func validRoleConstraints(role *roleStorageEntry) (bool, error) {
```

Comment:
It was that way, using just the error at first. I had to skip on checking role being nil in that case. I guess that's something we can live with. I'll revert it back to using just the error and punt on checking role for nil. And add a note in the method's comments.